src/topp/OpenPepXL.cpp

```diff
@@ -232,9 +232,26 @@ class TOPPOpenPepXL :
 
     OpenPepXLAlgorithm search_algorithm;
     Param this_param = getParam_().copy("", true);
-    Param algo_param = search_algorithm.getParameters();
-    algo_param.update(this_param, false, OpenMS_Log_debug); // suppress param. update message
-    search_algorithm.setParameters(algo_param);
+
+    // remove tool specific parameters before passing them to the algorithm
+    // to suppress warnings about unknown parameters
+    this_param.remove("in");
+    this_param.remove("consensus");
+    this_param.remove("database");
+    this_param.remove("decoy_database");
+    this_param.remove("out_idXML");
+    this_param.remove("out_mzIdentML");
+    this_param.remove("out_xquestxml");
+    this_param.remove("out_xquest_specxml");
+    this_param.remove("ini");
+    this_param.remove("log");
+    this_param.remove("debug");
+    this_param.remove("threads");
+    this_param.remove("no_progress");
+    this_param.remove("force");
```

Comment:
please don't do this.

this list is not tested against TOPPBase and its very verbose.

Better:
add a method to TOPPBase which returns a Param which contains the common parameters.
You need to convert between `Param` and `ParameterInformation`, but there is code for that: `TOPPBase::paramToParameterInformation_(const Param& param) `.

Reply:
ideally, you move the `TOPPBase::paramToParameterInformation_(const Param& param) ` into the `ParameterInformation` class, where it belongs while you are at it :-)

Reply:
other solution: find overlapping set by comparing default params of TOPP tool vs default params of algorithm.